projects/core/components/hosted-dropdown/hosted-dropdown.component.ts

```diff
@@ -159,7 +172,8 @@ export class TuiHostedDropdownComponent implements TuiFocusableElementAccessor {
         }
     }
 
-    @HostListener('keydown.esc', ['$event'])
+    @shouldCall(activeZoneFilter)
+    @HostListener('document:keydown.escape', ['$event'])
```

Comment:
Right now there's no point in this code. `@shouldCall` is used for filtering calls and **re-entering** the zone. We did not leave the zone, so all we did is write a very complex early return. We need `@HostListener('document:keydown.silent')` and inside `@shouldCall` we need to filter for escape key, because Angular re-enters zone if we rely on its built-in event manager plugin (`keydown.escape`). This can also be reused in `InputCardGrouped`.

Reply:
`Right now there's no point in this code. ` it would be great if there would be a eslint rule for that

Reply:
fixed